primitives/src/identity.rs

```diff
@@ -41,26 +41,33 @@ where
         }
     }
 
+    /// Check if `signer` is already a secondary key.
+    pub fn contains_secondary_key(&self, signer: &Signatory<AccountId>) -> bool {
+        self.secondary_keys.iter().any(|sk| sk.signer == *signer)
+    }
+
     /// It adds `new_secondary_keys` to `self`.
     /// It also keeps its internal list sorted and removes duplicate elements.
```

Comment:
This comment can be removed as the list is no longer sorted and duplicates shouldn't be possible with the new check
```suggestion

```
I assume there is no other reason to sort? I actually like that linked accounts will remain sorted in the order they were added rather than sorting by key.

Reply:
Thanks!  I forgot about that comment.